README.md

```diff
@@ -1,2 +1,203 @@
-# wp-cli-bulk-task
-A library to assist with running performant bulk tasks against posts via WP-CLI.
+# WP Bulk Task
+
+[![Readme Standard Spec Badge](https://img.shields.io/badge/readme%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
+
+A library to assist with running performant bulk tasks against WordPress posts.
+
+## Background
+
+This package provides a library to make it easier to run bulk tasks against a
+WordPress database in a performant way. It includes functionality to search
+through a WordPress database for posts using WP_Query-style arguments and keeps
+a cursor of its location within the database in case it is interrupted and needs
+to start again.
+
+
+## Releases
+
+This package is released via Packagist for installation via Composer. It follows
+semantic versioning conventions.
+
+
+### Install
+
+Requires Composer and PHP >= 8.0.
+
+
+### Use
+
+Install this package via Composer:
+
+```sh
+composer require alleyinteractive/wp-bulk-task
+```
+
+Ensure that the Composer autoloader is loaded into your project:
+
+```php
+require_once __DIR__ . '/vendor/autoload.php';
+```
+
+Then use the class in your custom CLI command:
+
+```php
+class My_Custom_CLI_Command extends WP_CLI_Command {
+
+	/**
+	 * Replace all instances of 'apple' with 'banana' in post content.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--dry-run]
+	 * : If present, no updates will be made.
+	 *
+	 * [--rewind]
+	 * : Resets the cursor so the next time the command is run it will start from the beginning.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp my-custom-cli-command bananaify
+	 *
+	 * @subcommand bananaify
+	 */
+	public function bananaify( $args, $assoc_args ) {
+		$bulk_task = new \Alley_Interactive\WP_Bulk_Task\Bulk_Task(
+			'bananaify',
+			new \Alley_Interactive\WP_Bulk_Task\Progress\PHP_CLI_Progress_Bar(
+				__( 'Bulk Task: remove_broken_links', 'my-textdomain' )
+			)
+		);
+
+		// Handle rewind requests.
+		if ( ! empty( $assoc_args['rewind'] ) ) {
+			$bulk_task->cursor->reset();
+			WP_CLI::log( __( 'Rewound the cursor. Run again without the --rewind flag to process posts.', 'my-textdomain' ) );
+			return;
+		}
```

Comment:
I went back and forth on this. I wanted the base class to not directly depend on WP_CLI, and I wanted the rewind functionality to be handled (or at least handleable) separately from the command running, because if it would get killed and restarted, with the `--rewind` flag it would start over every time, which defeats the point. Maybe in a future iteration we could create a WP_CLI helper that handles things like rewind, dry run, etc automatically, but keeps the main class "pure" so it can be used outside of a WP_CLI context?